project/v1.ts

```diff
@@ -107,28 +107,27 @@ class ProjectV1 extends BaseService {
    * the configuration. After the changes are validated and approved, deploy the resources that the project configures.
    *
    * @param {Object} params - The parameters to send to the service.
-   * @param {string} params.resourceGroup - The resource group where the project's data and tools are created.
-   * @param {string} params.location - The location where the project's data and tools are created.
-   * @param {string} params.name - The project name.
-   * @param {string} [params.description] - A project's descriptive text.
-   * @param {boolean} [params.destroyOnDelete] - The policy that indicates whether the resources are destroyed or not
-   * when a project is deleted.
-   * @param {ProjectConfigPrototype[]} [params.configs] - The project configurations.
+   * @param {ProjectPrototypeDefinition} params.definition - The definition of the project.
+   * @param {string} params.location - The IBM Cloud location where a resource is deployed.
+   * @param {string} params.resourceGroup - The resource group name where the project's data and tools are created.
+   * @param {ProjectConfigPrototype[]} [params.configs] - The project configurations. These configurations are only
+   * included in the response of creating a project if a configs array is specified in the request payload.
+   * @param {EnvironmentPrototype[]} [params.environments] - The project environments. These environments are only
+   * included in the response of creating a project if a environments array is specified in the request payload.
    * @param {OutgoingHttpHeaders} [params.headers] - Custom request headers
    * @returns {Promise<ProjectV1.Response<ProjectV1.Project>>}
    */
   public createProject(
     params: ProjectV1.CreateProjectParams
   ): Promise<ProjectV1.Response<ProjectV1.Project>> {
     const _params = { ...params };
-    const _requiredParams = ['resourceGroup', 'location', 'name'];
+    const _requiredParams = ['definition', 'location', 'resourceGroup'];
```

Comment:
It looks like there are a lot of breaking changes being made here...

Is there a reason you already a released a v1.0 even though the API is changing so much? If you'd stuck to a pre-release at first, you wouldn't have an issue here.

Reply:
In 6 month we changed a lot our API, but now we have no plans to make major changes in next months.
What do you suggest at this point?

Reply:
Because the code as is it won't work at all, I suppose it's fine to proceed with a "fix" level release here.

That said, try to capture the situation in your commit message as best you can so that any users coming to this repo to try and understand the changes have solid documentation to go off of. The documentation and UX of the package is only as good as you make it!